### DISCUSSION TITLE
구경가기 디벨롭

### DISCUSSION THREAD

**Discussion by Jaymyong66** [woowacourse-teams/2024-code-zap#801] [결정 사항]
구경가기에 기존 검색만 있던 부분을 개선한다

예를 들어, 상단에 배너를 두고 태그도 둔다.
상단에서는 큰 주제의 파트, 태그에는 구체적인 react, spring 등
이렇게 했을 때, 백엔드를 누르면 어떻게 쿼리를 보내고 보여줄 것인가?

안건 1) 기획 부분 제안 부탁드립니다. 
상단에 백엔드, 프론트엔드 이런 부분이 너무 크다는 의견도 있었어요.

안건 2) 백엔드 배너를 누르면 프론트에서 임의로 정한 java, spring  등의 태그를 넣어 쿼리를 보낸다?
하지만 태그를 설정 안하면 검색하기 어렵고, spring, Spring, 스프링 모두 검색해야한다.

안건 3) 프론트엔드의 경우, js, ts, jsx, tsx 확장자로 검색이 가능하다면? 
또 추후 다른 서비스(?)를 위해 확장자 테이블을 분리하자는 의견이 있었다

<예시>
<img width="1089" alt="스크린샷 2024-10-17 오전 10 06 30" src="https://github.com/user-attachments/assets/3a295742-bb64-48d5-9a67-7f88e68420d7">

<레퍼런스로 봤던 것>
<img width="981" alt="스크린샷 2024-10-17 오전 10 56 52" src="https://github.com/user-attachments/assets/b95077f2-fcbe-4bbe-91de-4eb7a3112a27">

**Comment by Jaymyong66** [comment]
## 어떻게 검색?
> 검색은 태그로 밖에 할 수 없다

모든 검색은 태그 기반으로

## 올라가는 태그는 어떻게 선택?

> 프론트, 백엔드 같은 것은 안함

java, javascript 등 고정된 주제 +  인기순 선정

## 태그 검색 결과를 늘리기 위해서는?

1. 확장자에 따라 언어 태그를 자동 생성해준다
2. 대소문자 구분을 없앤다

**Reply by Jaymyong66** [reply]
자바, 자바스크립트, 코틀린, 우테코, 프리코스, 알고리즘, 리액트, 스프링, 안드로이드

**Reply by kyum-q** [reply]
## Tag id 목록

### 자바 (자바, java)
73, 358, 197

### 자바 스크립트 (자바스크립트, javascript, js)
41, 211, 329, 351, 249, 360

### 코틀린 (코틀린, kt)
361, 363

### 우테코 
359

### 프리코스
364

### 알고리즘 (알고리즘, 코테, 코딩테스트, BFS, DFS)
261, 316, 253, 144, 143

### 리액트 (리액트, react)
50, 289, 318

### 스프링 (스프링, spring)
14, 198

### 안드로이드 
362

**Reply by kyum-q** [reply]
kotlin - 237
android - 236, 287

**Comment by Jaymyong66** [comment]
### 회의 후 해야할 일
헤인 - 유튜브 영상, 주말간
월하 - 홍보글 작성, 월요일까지

### 프론트 
- 태그 자동 생성
- 배너 : 자바, 자바스크립트, 코틀린, 우테코, 프리코스, 알고리즘, 리액트, 스프링, 안드로이드


### 백엔드 
- OR로 바꾸기
- 태그 ID 알려주기

### 추후 개선 사항
- 확장자 테이블 분리하기
